savoirfairelinux/sous-chef
# Fixed #691: new "Route" menu for viewing and editing delivery sequences

## Fixes #691 by lingxiaoyang

### Changes proposed in this pull request:

* New model `member.DeliveryHistory`: stores the specific vehicle/client sequence/comments of routes on each delivery day.
* New main menu "Routes": provides list/detail/edit view of a `member.Route`, notably allows the editing of default delivery sequence.
* Logic change in kitchen count -> organize route: works on `member.DeliveryHistory` instead of `member.Route`. The operation "organize route" creates the `member.DeliveryHistory`. The route sheet is only available after having today's `DeliveryHistory`.
* Frontend: refactored leaflet map code, concerning (a) client creation/edit form, (b) route view/edit, (c) delivery route edit.
* Basic unit tests.


### Status


### How to verify this change

New URL: try updating and viewing the routes.
http://localhost:8000/member/routes/

Verify the leaflet map in client creation/edit form works well.

Process kitchen count, in last step, try organising routes. After that, go back to "Route" menu and you will find delivery histories.

### Deployment notes and migration


### New translatable strings


### Additional notes

none
